File: Lab7/Question5/src/Magpie.java
public class Magpie extends Bird {
    public Magpie(String name) {
        super(name, 6);
    }
    
    @Override
    public boolean canFly() {
    	return true;
    }
    public static void testMagpie() {
        Magpie magpie = new Magpie("Minami");
        System.out.println(magpie.getName() == "Minami");
        System.out.println(magpie.getLegs() == 2);
        System.out.println(magpie.getNumOfEggs() == 6);
        System.out.println(magpie.canFly() == true);
    }
}

File: Lab7/Magpie.kt
class Magpie : Bird {
    constructor(name: String) : super(name, 6)

    override fun canFly(): Boolean {
        return true
    }

    companion object {
        fun testMagpie() {
            val magpie = Magpie("mag")
            println(magpie.getNumOfEggs() == 6)
            println(magpie.getName() == "mag")
            println(magpie.getLegs() == 2)
            println(magpie.isDangerous() == false)
            println(magpie.canFly() == true)
            val mag = Magpie("mp")
            println(mag.getName() == "mp")
            println(mag.isDangerous() == false)
            println(mag.canFly() == true)
        }
    }
}